src/Menu/Menu.js

```diff
@@ -125,10 +125,6 @@ class Menu extends Component<DefaultProps, AllProps, void> {
       // $FlowFixMe
       menuList.style.width = `calc(100% + ${size})`;
     }
-
-    if (this.props.onEnter) {
-      this.props.onEnter(element);
```

Comment:
The why is around consistency with the rest of our API, it would increase the entropy. Instead, we could imagine adding a property to disable the default behavior, it would make it explicit, much better :).